polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -173,6 +173,11 @@ MLIRScanner::VisitImplicitValueInitExpr(clang::ImplicitValueInitExpr *Decl) {
     return ValueCategory(Builder.create<mlir::LLVM::NullOp>(Loc, PT), false,
                          Builder.getI8Type());
 
+  if (auto PT = dyn_cast<mlir::LLVM::LLVMTargetExtType>(MLIRTy))
+    return ValueCategory(Builder.create<mlir::LLVM::ConstantOp>(
+                             Loc, PT, Builder.getI64IntegerAttr(0)),
+                         false, Builder.getI8Type());
```

Comment:
Can we get a lit test for this?

Reply:
From which stage, i.e., should the input be C++ or MLIR?

Reply:
c++ to mlir

Reply:
C++ to MLIR, yep

Reply:
Turns out we already had multiple tests for that were marked `XFAIL` during the initial work on opaque pointers. I re-enabled them and updated them to work correctly with opaque pointer.